src/module.c

```diff
@@ -1671,21 +1689,32 @@ static int moduleValidateCommandInfo(const RedisModuleCommandInfo *info) {
     return moduleValidateCommandArgs(info->args, version);
 }
 
-/* Converts from REDISMODULE_CMD_KEY_* flags to CMD_KEY_* flags. */
-static int64_t moduleConvertKeySpecsFlags(int64_t flags) {
-    int64_t realflags = 0;
-    if (flags & REDISMODULE_CMD_KEY_RO) realflags |= CMD_KEY_RO;
-    if (flags & REDISMODULE_CMD_KEY_RW) realflags |= CMD_KEY_RW;
-    if (flags & REDISMODULE_CMD_KEY_OW) realflags |= CMD_KEY_OW;
-    if (flags & REDISMODULE_CMD_KEY_RM) realflags |= CMD_KEY_RM;
-    if (flags & REDISMODULE_CMD_KEY_ACCESS) realflags |= CMD_KEY_ACCESS;
-    if (flags & REDISMODULE_CMD_KEY_INSERT) realflags |= CMD_KEY_INSERT;
-    if (flags & REDISMODULE_CMD_KEY_UPDATE) realflags |= CMD_KEY_UPDATE;
-    if (flags & REDISMODULE_CMD_KEY_DELETE) realflags |= CMD_KEY_DELETE;
-    if (flags & REDISMODULE_CMD_KEY_CHANNEL) realflags |= CMD_KEY_CHANNEL;
-    if (flags & REDISMODULE_CMD_KEY_INCOMPLETE) realflags |= CMD_KEY_INCOMPLETE;
-    if (flags & REDISMODULE_CMD_KEY_VARIABLE_FLAGS) realflags |= CMD_KEY_VARIABLE_FLAGS;
-    return realflags;
+/* When from_api is true, converts from REDISMODULE_CMD_KEY_* flags to CMD_KEY_* flags.
+ * When from_api is false, converts from CMD_KEY_* flags to REDISMODULE_CMD_KEY_* flags. */
+static int64_t moduleConvertKeySpecsFlags(int64_t flags, int from_api) {
+    int64_t out = 0;
+    int64_t map[][2] = {
+        {REDISMODULE_CMD_KEY_RO, CMD_KEY_RO},
+        {REDISMODULE_CMD_KEY_RW, CMD_KEY_RW},
+        {REDISMODULE_CMD_KEY_OW, CMD_KEY_OW},
+        {REDISMODULE_CMD_KEY_RM, CMD_KEY_RM},
+        {REDISMODULE_CMD_KEY_ACCESS, CMD_KEY_ACCESS},
+        {REDISMODULE_CMD_KEY_INSERT, CMD_KEY_INSERT},
+        {REDISMODULE_CMD_KEY_UPDATE, CMD_KEY_UPDATE},
+        {REDISMODULE_CMD_KEY_DELETE, CMD_KEY_DELETE},
+        {REDISMODULE_CMD_KEY_CHANNEL, CMD_KEY_CHANNEL},
+        {REDISMODULE_CMD_KEY_INCOMPLETE, CMD_KEY_INCOMPLETE},
+        {REDISMODULE_CMD_KEY_VARIABLE_FLAGS, CMD_KEY_VARIABLE_FLAGS},
+        {0,0}};
+    if (from_api) {
+        for (int i=0; map[i][0]; i++)
+            if (flags & map[i][0]) out |= map[i][1];
+    } else {
+        for (int i=0; map[i][0]; i++)
+            if (flags & map[i][1]) out |= map[i][0];
+    }
```

Comment:
Avoid repeating the loop? (perhaps a matter of taste)

```suggestion
    int from_idx = from_api ? 0 : 1, to_idx = !from_idx;
    for (int i=0; map[i][0]; i++)
        if (flags & map[i][from_idx]) out |= map[i][to_idx];
```